From fb01d7d042818a2b83e1a4b2354cf636606e9e21 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Wed, 15 Oct 2014 13:44:22 -0400
Subject: [crypto] qat: Prevent dma mapping zero length assoc data

Message-id: <1413380663-25484-2-git-send-email-nikolay@redhat.com>
Patchwork-id: 97574
O-Subject: [RHEL7.1 PATCH BZ 1152642 1/2] crypto: qat - Prevent dma mapping zero length assoc data
Bugzilla: 1152642
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Tadeusz Struk <tadeusz.struk@intel.com>

Do not attempt to dma map associated data if it is zero length.

Signed-off-by: Tadeusz Struk <tadeusz.struk@intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/crypto/qat/qat_common/qat_algs.c b/drivers/crypto/qat/qat_common/qat_algs.c
index 59df488..5b158ca 100644
--- a/drivers/crypto/qat/qat_common/qat_algs.c
+++ b/drivers/crypto/qat/qat_common/qat_algs.c
@@ -650,6 +650,8 @@ static int qat_alg_sgl_to_bufl(struct qat_crypto_instance *inst,
   goto err;
 
  for_each_sg(assoc, sg, assoc_n, i) {
+  if (!sg->length)
+   continue;
   bufl->bufers[bufs].addr = dma_map_single(dev,
         sg_virt(sg),
         sg->length,
-- 
1.7.1